src/main/scala/edu/ie3/simona/agent/grid/DBFSAlgorithm.scala

```diff
@@ -278,61 +279,108 @@ trait DBFSAlgorithm extends PowerFlowSupport with GridResultsSupport {
+          )
         ) =>
-      log.debug(
-        "Received request for grid power values and im READY to provide."
+      /* Determine the subnet number of the grid agent, that has sent the request */
+      val firstRequestedNodeUuid = requestedNodeUuids.headOption.getOrElse(
+        throw new DBFSAlgorithmException(
+          "Did receive a grid power request but without specified nodes"
+        )
       )
+      gridAgentBaseData.gridEnv.subnetGateToActorRef
+        .map { case (subGridGate, _) => subGridGate.getSuperiorNode }
+        .find(_.getUuid == firstRequestedNodeUuid)
+        .map(_.getSubnet) match {
+        case Some(requestingSubnetNumber) =>
+          log.debug(
+            "Received request for grid power values and im READY to provide."
+          )
 
-      // get the index from the power flow data of the requested node
-      val requestedNodeIdxOpt =
-        gridAgentBaseData.gridEnv.gridModel.nodeUuidToIndexMap
-          .get(requestedNodeUuid)
-
-      powerFlowResult match {
-        case validNewtonRaphsonPFResult: ValidNewtonRaphsonPFResult =>
-          val (p, q) = validNewtonRaphsonPFResult.nodeData.find(stateData =>
-            requestedNodeIdxOpt.contains(stateData.index)
-          ) match {
-            case Some(apparentPowerS) =>
-              val refSystem = gridAgentBaseData.gridEnv.gridModel.mainRefSystem
-              val (pInPu, qInPu) =
-                (apparentPowerS.power.real, apparentPowerS.power.imag)
-              // The power flow result data provides the nodal residual power at the slack node.
-              // A feed-in case from the inferior grid TO the superior grid leads to positive residual power at the
-              // inferior grid's *slack node* (superior grid seems to be a load to the inferior grid).
-              // To model the exchanged power from the superior grid's point of view, -1 has to be multiplied.
-              // (Inferior grid is a feed in facility to superior grid, which is negative then). Analogously for load case.
-              (
-                refSystem.pInSi(pInPu).multiply(-1),
-                refSystem.qInSi(qInPu).multiply(-1)
-              )
-            case None =>
-              throw new DBFSAlgorithmException(
-                s"Got a request for power @ node with uuid $requestedNodeUuid but cannot find it in my result data!"
-              )
-          }
+          powerFlowResult match {
+            case validNewtonRaphsonPFResult: ValidNewtonRaphsonPFResult =>
+              val exchangePowers = requestedNodeUuids
+                .map { nodeUuid =>
+                  /* Figure out the node index for each requested node */
+                  nodeUuid -> gridAgentBaseData.gridEnv.gridModel.nodeUuidToIndexMap
+                    .get(nodeUuid)
+                    .flatMap { nodeIndex =>
+                      /* Find matching node result */
+                      validNewtonRaphsonPFResult.nodeData.find(stateData =>
```

Comment:
It's a nice idea, we'd just have to realize in [powerflow](https://github.com/ie3-institute/powerflow), as that's where the `PowerFlowResult` classes reside

Reply:
Created an issue here: https://github.com/ie3-institute/powerflow/issues/65

Reply:
I now did a very similar thing with `SweepValueStore` though... :)

Reply:
Ok regarding `SweepValueStore`, I reverted my changes because StateData needs to be in a certain order (Maps don't preserve order) and I can't figure out where the problem is currently. It'll be a problem for another day, I think in total DBFS can use some more revisions anyways.